segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java

```diff
@@ -50,39 +52,39 @@ public SegmentTruncatedException(String message) {
      * @param targetOffset The requested offset's corresponding position to search in the index segment entry.
      * @param greater boolean to determine if the next higher or the lower value to be returned in case the requested offset is not present.
      *
-     * @return the corresponding offset position from the index segment entry.
+     * @return A future for the corresponding offset position from the index segment entry.
      * @throws SegmentTruncatedException If the segment is truncated.
      */
-    public static long findNearestIndexedOffset(StreamSegmentStore store, String segment, long targetOffset,
-                                                boolean greater) throws SegmentTruncatedException {
+    public static CompletableFuture<Long> findNearestIndexedOffset(StreamSegmentStore store, String segment,
+                                                                   long targetOffset,
+                                                                   boolean greater) throws SegmentTruncatedException {
         String indexSegmentName = NameUtils.getIndexSegmentName(segment);
 
         // Fetch start and end idx.
         SegmentProperties properties = store.getStreamSegmentInfo(indexSegmentName, TIMEOUT).join();
```

Comment:
Can we remove join() from here? Can we use **store.getStreamSegmentInfo** in same chain like 
store.getStreamSegmentInfo(indexSegmentName, TIMEOUT).thenCompose(x -> {}).